### PR TITLE
feat(java): add `test` scope support for `pom.xml` files

### LINKED ISSUE #7384: feat(java): detect and mark as `Dev` dependencies with `test` scope from `pom.xml` files
## Description
Define dependencies with [test scope](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope) and mark them as `Dev` for show using `--include-dev-deps` flag.

### Discussed in https://github.com/aquasecurity/trivy/discussions/7346

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add `test` scope support and mark these dependencies as `Dev`.
To show these deps - use `--include-dev-deps` flag.

## Related issues
- Close #7384

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by knqyf263 — APPROVED** [review]

**Review by knqyf263 — COMMENTED** [review]
Do we need to update the document?

**Comment by DmitriyLewen** [comment]
Yeah. You are right. Thanks!

Add info about scopes in https://github.com/aquasecurity/trivy/pull/7414/commits/e3cfe7231d8ecc405ffde162172bbee75e8d9777

**Review by knqyf263 — COMMENTED** [review]
I think the table also needs to be updated like [npm](https://aquasecurity.github.io/trivy/v0.54/docs/coverage/language/nodejs).
https://github.com/aquasecurity/trivy/blob/e3cfe7231d8ecc405ffde162172bbee75e8d9777/docs/docs/coverage/language/java.md?plain=1#L18

**Comment by DmitriyLewen** [comment]
You are right again.
Updated in https://github.com/aquasecurity/trivy/pull/7414/commits/a162fa15ba2415e7c575cbe0f317ec0ad8e90e1b

File: docs/docs/coverage/language/java.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ Each artifact supports the following scanners:
...
-| pom.xml          | Maven repository [^1] |     Exclude      |                  ✓                   |  ✓[^7]   |                    -                     |
-| *gradle.lockfile |           -           |     Exclude      |                  ✓                   |    ✓     |                Not needed                |
-| *.sbt.lock       |           -           |     Exclude      |                  -                   |    ✓     |                Not needed                |
+| Artifact         |    Internet access    |  Dev dependencies  | [Dependency graph][dependency-graph] | Position | [Detection Priority][detection-priority] |
+|------------------|:---------------------:|:------------------:|:------------------------------------:|:--------:|:----------------------------------------:|
+| JAR/WAR/PAR/EAR  |     Trivy Java DB     |      Include       |                  -                   |    -     |                Not needed                |
+| pom.xml          | Maven repository [^1] | [Include](#scopes) |                  ✓                   |  ✓[^7]   |                    -                     |
```
Comment (knqyf263) [inline docs/docs/coverage/language/java.md:18]: You changed from "Exclude" to "Include", but IMO, it should describe the default behavior. Actually, Node.js says "Excluded" although it supports `--include-dev-deps`. If you think "Include" is better, we need to change Node.js and other ecosystems for consistency.

```suggestion
| pom.xml          | Maven repository [^1] | [Exclude](#scopes) |                  ✓                   |  ✓[^7]   |                    -                     |
```

Comment (DmitriyLewen) [reply docs/docs/coverage/language/java.md]: Hmm… 
This is a very controversial situation, but I would probably agree with you that we should specify the default behavior in this table.
Updated in https://github.com/aquasecurity/trivy/pull/7414/commits/117555f15ec0d6daacff5dfdcd2f83b368364658

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**